crates/wasi/src/snapshots/preview_1.rs

```diff
@@ -0,0 +1,151 @@
+#[allow(unused_imports)]
+use wasi_common::snapshots::preview_1::*;
+use wasi_common::Error;
+use wasmi::{core::Trap, AsContextMut, Caller, Extern, Func, Linker};
+use wasmi_core::HostErrType;
+
+/// Adapted from wasmtime's `wiggle` crate. To reuse this would have required
+/// an implementation of `From<wasmtime::Trap>`
+fn run_in_dummy_executor<F: std::future::Future>(future: F) -> Result<F::Output, Trap> {
```

Comment:
This whole function with its mentions of an `async Store` sounds too much of a copy&paste from Wasmtime since there is no notion of `async` with respect to `wasmi`. Also I have to state that I currently do not really understand what we need this for exactly and why those `dummy_waker` functions are implemented the way they are. More docs needed here for me.